packages/react-reconciler/src/ReactFiberHooks.old.js

```diff
@@ -1498,25 +1503,43 @@ function rerenderDeferredValue<T>(
 
 function startTransition(setPending, config, callback) {
   const priorityLevel = getCurrentPriorityLevel();
-  runWithPriority(
-    priorityLevel < UserBlockingPriority ? UserBlockingPriority : priorityLevel,
-    () => {
-      setPending(true);
-    },
-  );
-  runWithPriority(
-    priorityLevel > NormalPriority ? NormalPriority : priorityLevel,
-    () => {
-      const previousConfig = ReactCurrentBatchConfig.suspense;
-      ReactCurrentBatchConfig.suspense = config === undefined ? null : config;
-      try {
-        setPending(false);
-        callback();
-      } finally {
-        ReactCurrentBatchConfig.suspense = previousConfig;
-      }
-    },
-  );
+  const previousLanePriority = getCurrentLanePriority();
+  try {
+    setCurrentLanePriority(
+      priorityLevel < UserBlockingPriority
+        ? InputContinuousLanePriority
+        : schedulerPriorityToLanePriority(priorityLevel),
+    );
+    runWithPriority(
+      priorityLevel < UserBlockingPriority
+        ? UserBlockingPriority
+        : priorityLevel,
+      () => {
+        setPending(true);
+      },
+    );
+
+    setCurrentLanePriority(
+      priorityLevel > NormalPriority
+        ? DefaultLanePriority
+        : schedulerPriorityToLanePriority(priorityLevel),
```

Comment:
This should be one of the transition priorities. Short or long, depending on the timeout. Here's where that logic currently exists: https://github.com/facebook/react/blob/8f05f2bd6d131a39835d468622e248b231ccbf8e/packages/react-reconciler/src/ReactFiberWorkLoop.new.js#L402-L406

Reply:
Should the priority on line 1509 also use the short/long lane priorities or are those fine?

Reply:
That one should always be InputContinuousLanePriority.

Reply:
Are you sure about this? The transition lanes don't have an update lane: 

https://github.com/rickhanlonii/react/blob/0993d6dad6303e13098a3bf4304332ad2f67b546/packages/react-reconciler/src/ReactFiberLane.js#L538

Reply:
Changing to the transition lane fails a couple tests for: 

<img width="803" alt="Screen Shot 2020-06-12 at 3 34 47 PM" src="https://user-images.githubusercontent.com/2440089/84539890-4cc78b00-acc2-11ea-9611-a2dc2ba7bea5.png">

Reply:
See the comment on line 538. Since this is a transition, there should be a SuspenseConfig object, and so `requestUpdateLane` should call `findTransitionLane` instead of `findUpdateLane`:

https://github.com/facebook/react/blob/103ed08c46198d01119ef35c37d78c6bc89705db/packages/react-reconciler/src/ReactFiberWorkLoop.new.js#L398-L419

There might be a better way to factor that, but since we currently still need the SuspenseConfig for timeouts, anyway, this should work for now.

Reply:
The tests are failing because there's no SuspenseConfig object. How would you expect the SuspenseConfig to be set (e.g. for [this test](https://github.com/facebook/react/blob/fe7163e73dadceda2655736d97cdd745d7abc8ea/packages/react-reconciler/src/__tests__/ReactTransition-test.js#L62))?